### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by YahyoMahmudov**

### CODE REVIEW

File: task1/Calculator.test.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { Calculator } from "./Calculator";
+
+var calculator = new Calculator();
```
Comment (jkalandarov) [inline task1/Calculator.test.ts]: I recommend using `let` or `const` over `var`.

File: task1/Calculator.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+export class Calculator {
+
+    addition(x, y): number {
+
+        if (typeof x !== 'number' || typeof y !== 'number') {
+            return NaN;
```
Comment (jkalandarov) [inline task1/Calculator.ts]: In case of invalid type, it's better to throw an error.

File: task2/DivisibleBy.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { divisibleBy } from "./DivisibleBy";
+
+test("Given number is divisible by 3 and not 5", () => {
+    expect(divisibleBy(3)).toBe(false);
+})
```
Comment (jkalandarov) [inline task2/DivisibleBy.test.ts:5]: Are you checking if a given number is divisible by 3? if so, 3 is divisible by 3.

File: task2/DivisibleBy.test.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+test("Given number is divisible by 5 and not 3", () => {
+    expect(divisibleBy(5)).toBe(false);
+})
```
Comment (jkalandarov) [inline task2/DivisibleBy.test.ts:9]: Are you checking if a given number is divisible by 5? if so, 5 is divisible by 5.

File: task2/DivisibleBy.test.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+test("Given number is '5'", () => {
+    expect(divisibleBy('5')).toBe(false);
+})
+
+test("Given number is '15'", () => {
+    expect(divisibleBy('15')).toBe(false);
+})
+
```
Comment (jkalandarov) [inline task2/DivisibleBy.test.ts]: The test names are confusing.

File: task3/ArrayParam.test.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { arrayParam } from "./ArrayParam";
+
+test("Given object's age is above or equal to 18", () => {
+    expect(arrayParam([ {name:"John",age:22}, {name:"Jack",age:18}, {name:"James",age:40} ])).toBe(true);
+})
+
+test("Given object's age is not above or equal to 18", () => {
+    expect(arrayParam([ {name:"John",age:10}, {name:"Jack",age:32}, {name:"James",age:40} ])).toBe(false);
+})
```
Comment (jkalandarov) [inline task3/ArrayParam.test.ts]: John's age is below 18. According to the test name, the result should be `true`.

File: task3/ArrayParam.test.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+test("Given object's age is not above or equal to 18", () => {
+    expect(arrayParam([ {name:"John",age:18}, {name:"Jack",age:17}, {name:"James",age:19} ])).toBe(false);
+})
```
Comment (jkalandarov) [inline task3/ArrayParam.test.ts]: Jack's age is not above or equal to 18. The result should be `true`.

File: task3/ArrayParam.test.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+test("Given object's age is not above or equal to 18", () => {
+    expect(arrayParam([ {name:"John",age:2}, {name:"Jack",age:1}, {name:"James",age:0} ])).toBe(false);
+})
```
Comment (jkalandarov) [inline task3/ArrayParam.test.ts]: None of the object's age is not above or equal to 18, the result should be `true` for all cases.

File: task3/ArrayParam.test.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+test("Given object's age is above or equal to 18", () => {
+    expect(arrayParam([ {name:"John",age:18}, {name:"Jack",age:38}, {name:"James",age:58} ])).toBe(true);
+})
```
Comment (jkalandarov) [inline task3/ArrayParam.test.ts]: What if one of the object's age is below 18? How do you handle such cases?

File: task3/ArrayParam.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+export function arrayParam(arr: { name: string, age: number }[]): boolean {
```
Comment (jkalandarov) [inline task3/ArrayParam.ts]: function's name should be a verb and tell what this function does. 
Check these [naming conventions](https://www.syncfusion.com/blogs/post/top-javascript-naming-convention.aspx) in programming.

File: task4/DateFormat.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
+export function dateFormat(strDate: string): boolean {
+    
+    // ****** Using Date class
+
+    var date = new Date(strDate);
+    return !isNaN(date.getTime());
+
+
+
```
Comment (jkalandarov) [inline task4/DateFormat.ts]: Don't leave extra spaces if there's no need for them.

File: task4/DateFormat.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+    // ****** Using regex
+
+    /* var regex = /^(0[1-9]|[12][0-9]|3[01])-(0[1-9]|1[0-2])-\d{4}$|^(\d{4})-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$|^((0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])-\d{4})$/;
+    return regex.test(strDate); */
+
+
```
Comment (jkalandarov) [inline task4/DateFormat.ts]: It's recommended not to leave unnecessary comments.

File: task4/DateFormat.ts [hunk 13]
```diff
@@ -0,0 +1,16 @@
+export function dateFormat(strDate: string): boolean {
+    
+    // ****** Using Date class
+
```
Comment (jkalandarov) [inline task4/DateFormat.ts]: Your code should be self documenting and should not leave a need for comments.

**Review by jkalandarov — CHANGES_REQUESTED** [review]

File: task2/DivisibleBy.test.ts (hunk 3 above)
Comment (YahyoMahmudov) [reply task2/DivisibleBy.test.ts:5]: I am checking that the number is divisible by both 3 and 5, not just one of them.

File: task3/ArrayParam.test.ts (hunk 6 above)
Comment (YahyoMahmudov) [reply task3/ArrayParam.test.ts]: According to the task description, all people's age should be 18 or above. Am I right?

File: task2/DivisibleBy.test.ts (hunk 4 above)
Comment (YahyoMahmudov) [reply task2/DivisibleBy.test.ts:9]: I am checking that the number is divisible by both 3 and 5, not just one of them.

File: task3/ArrayParam.test.ts (hunk 9 above)
Comment (YahyoMahmudov) [reply task3/ArrayParam.test.ts]: If one age is below 18, it should be false.

File: task3/ArrayParam.test.ts (hunk 8 above)
Comment (YahyoMahmudov) [reply task3/ArrayParam.test.ts]: Does the "above 18" mean higher than 18 or lower than 18? I think it is higher than 18. So, this test should be false.

File: task3/ArrayParam.test.ts (hunk 7 above)
Comment (YahyoMahmudov) [reply task3/ArrayParam.test.ts]: According to the description, if one of them is above 18, it should be false.

**Review by jkalandarov — APPROVED** [review]
Good job!

**Pull request merged**